web/src/main/webapp/WEB-INF/pages/application_state.jsp

```diff
@@ -106,53 +133,156 @@
                           <div class="col-sm-8">
                               <p class="form-control-static <c:if test="${stat.first}">stateData"</c:if> id="offeredMoneyP">${data.offeredMoney}</p>
                               <c:if test="${stat.first}">
-                                  <input class="stateInput" type="text" name="offeredMoney" id="offeredMoneyInput" style="display:none" value="${data.offeredMoney}">
+                                  <input class="stateInput" type="number" name="offeredMoney" id="offeredMoneyInput" style="display:none" value="${data.offeredMoney}"
+                                  data-error="<spring:message code="statehistory.error.offeredMoney.negative"/>"
+                                  data-bind="valueWithInit: 'offeredMoney'"
+                                  min="0">
                               </c:if>
                           </div>
+                          <div class="help-block with-errors"></div>
                       </div>
                       <div class="form-group">
                           <label for="name" class="control-label col-sm-4"><spring:message code="statehistory.field.claim"/></label>
                           <div class="col-sm-8">
                               <p class="form-control-static <c:if test="${stat.first}">stateData"</c:if> id="claimP">${data.claim}</p>
                               <c:if test="${stat.first}">
-                                  <input class="stateInput" type="text" name="claim" id="claimInput" style="display:none" value="${data.claim}">
+                                  <input class="stateInput" type="number" name="claim" id="claimInput" style="display:none" value="${data.claim}"
+                                  data-error="<spring:message code="statehistory.error.claim.negative"/>"
+                                  data-bind="valueWithInit: 'claim'"
+                                  min="0">
                               </c:if>
                           </div>
+                          <div class="help-block with-errors"></div>
                       </div>
                       <div class="form-group">
                           <label for="name" class="control-label col-sm-4"><spring:message code="statehistory.field.feedbackDate"/></label>
                           <div class="col-sm-8">
                               <p class="form-control-static <c:if test="${stat.first}">stateData"</c:if> id="feedbackDateP">${data.feedbackDate}</p>
                               <c:if test="${stat.first}">
-                                  <input class="stateInput" type="text" name="feedbackDate" id="feedbackDateInput" style="display:none" value="${data.feedbackDate}">
+                                  <input class="stateInput" type="text" name="feedbackDate" id="feedbackDateInput" style="display:none" value="${data.feedbackDate}"
```

Comment:
Do not use inline styles. Bootstrap has helper css classes, which can be used here, namely the "hidden" class.
Please see example in the master branch.